modules/node-shared/src/main/scala/org/tessellation/node/shared/infrastructure/snapshot/GlobalSnapshotStateChannelEventsProcessor.scala

```diff
@@ -89,16 +87,33 @@ object GlobalSnapshotStateChannelEventsProcessor {
           .flatMap { case (_, validatedEvents) => processStateChannelEvents(snapshotOrdinal, lastGlobalSnapshotInfo, validatedEvents) }
           .flatMap {
             case (scSnapshots, returnedSCEvents) =>
-              processCurrencySnapshots(lastGlobalSnapshotInfo, scSnapshots)
-                .map(calculateLastCurrencySnapshots(_, lastGlobalSnapshotInfo))
-                .map((scSnapshots, _, returnedSCEvents))
+              processCurrencySnapshots(snapshotOrdinal, lastGlobalSnapshotInfo, scSnapshots).map { accepted =>
```

Comment:
`process` and `processCurrencySnapshots` functions are were the core changes happen - please review thoroughly

Reply:
We know do final acceptance of binary and currency snapshot in the same function whereas previously we were accepting binaries separately. Accepting binaries separately wasn't correct in the first place - if a binary is a currency snapshot that can't be accepted why would we accept it as a binary - but currently we can also only accept custom binaries (non currency) if the fee is not required as we don't have a way to fetch a fee deduction address from custom binary.